bundle/jsky.app.ot/src/main/java/jsky/app/ot/viewer/SPViewer.java

```diff
@@ -887,6 +892,9 @@ public void closeProgram(ISPProgram node) {
 
     /** Close the current viewer window, but don't exit the application, even if it is the last window. */
     private void closeViewer() {
+        if (getRoot() != null)
+            BagsManager.instance().unwatch(getRoot());
```

Comment:
You can open more than a single program in a viewer.  When the viewer is closed, the cleanup should include all the programs in that viewer.  Can you confirm that this is happening as required?  It may be working fine but from this it looks like it could just be removing the active program in the view.